toolkit/scripts/download-packages.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash -e
+
+function get_packages {
+    # First argument is an URL to a directory containing packages
+    download_url="$1"
+
+    # wget -nv -O - "$download_url"               -- Download HTML of package directory and send to stdout (-nv reduces verbosity)
+    # | grep 'a href'                             -- Filter out lines that are not links (we're looking for links to rpms)
+    # | sed -E -e 's:<a href="(.+[^\])".+:\1:'    -- Remove everything but the address part of links (these are already URL encoded)
+    # | grep -v '/'                               -- '/' is invalid in RPM name but appears in links to different directories. Filter these out.
+    # | xargs -I {} wget -nv "$download_url"/{}   -- Download the packages
+
+    echo "-- Downloading packages from $download_url."
+    SECONDS=0
+    wget -nv -O - "$download_url" | grep 'a href' | sed -E -e 's:<a href="(.+[^\])".+:\1:' | grep -v '/' | xargs -P8 -I {} wget -nv "$download_url"/{}
+    echo "-- Finished downloading packages from $download_url. Operation took $SECONDS seconds."
+}
+
+function make_tarball {
+    archive_name=rpms.tar.gz
+
+    for package_type in $packages_types; do
+        mkdir -p RPMS/$package_type
+        mv *.$package_type.rpm RPMS/$package_type/
+    done
+
+    mkdir -p RPMS/noarch
+    mv *.noarch.rpm RPMS/noarch/
+
+    echo "-- Packaging into a tarball..."
+    tar --remove-files -czvf $archive_name RPMS
+}
+
+function help {
+    echo "Package downloader. Downloads packages from a repository."
+    echo "Usage:"
+    echo '[MANDATORY] -d DIR -> space-separated list of directories (e.g. "base update")'
```

Comment:
>space-separated list of directories (e.g. "base update")' [](http://example.com/codeflow?start=32&length=57)

Could you extend this description to explain what are these directories? From the code I'm guessing it's part of the repo's path to where the packages are stored.